api/xds_protocol.rst

```diff
@@ -469,3 +469,155 @@ occurred via a :ref:`resource update <xds_protocol_resource_update>`.
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+When an Envoy instance forms a delta xDS connection to the xDS management
+server for the first time, it will receive a base configuration filtered down
+to a subset of routes that are likely useful to the Envoy instance. For
+example, these might be filtered down by routes that exist in the current
+namespace. The xDS management server will send this same collection of
+resources on reconnect.
+
+.. figure:: diagrams/delta_rds_connection.svg
+   :alt: Delta RDS connection
+
+Virtual Host Resource Naming Convention
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Virtual hosts in VHDS are identified by a combination of the name of the route
+configuration to which the virtual host belong as well as the host HTTP "host"
+header (authority for HTTP2) entry. Resources should be named as follows:
+
+<route configuration name>/<host entry>
+
+Note that matching should be done from right to left since a host entry cannot
+contain slashes while a route configuration name can. 
+
+Requesting Additional resources
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+Envoy will send a
+:ref:`DeltaDiscoveryRequest <envoy_api_msg_DeltaDiscoveryRequest>` with the
+:ref:`resource_names_subscribe <envoy_api_field_DeltaDiscoveryRequest.resource_names_subscribe>`
+field populated with the virtual host resource names of each of the resources
+that it would like to subscribe to. Each of the virtual hosts contained in the
+:ref:`DeltaDiscoveryRequest's <envoy_api_msg_DeltaDiscoveryRequest>`
+resources field will be added to the route configuration maintained by Envoy.
+If Envoy's route configuration already contains a given virtual host, it will
+be overwritten by data received in the
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+but only if the updated virtual host is different from its current state.
+The :ref:`resource.aliases <envoy_api_field_resource.aliases>` field contains
+all host/authority header values used to create the on-demand request. The
+configuration server needs to keep track of virtual hosts known to Envoy.
+
+If a virtual host is requested for which the management server does not know
+about, then the management server should respond with a 
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>` in which
+the :ref:`resources <envoy_api_field_DeltaDiscoveryResponse.resources>` entry
+for that virtual host has the :ref:`name <envoy_api_field_resource.name>` and
+:ref:`aliases <envoy_api_field_resource.aliases>` set to the requested host
+entry and the resource unpopulated. This will allow Envoy to match the
+requested resource to the response.
+
+Envoy should respond to the 
+:ref:`DeltaDiscoveryResponse <envoy_api_msg_DeltaDiscoveryResponse>`
+with an Ack/Nack 
+:ref:`DeltaDiscoveryRequest's <envoy_api_msg_DeltaDiscoveryRequest>`.
+
+.. figure:: diagrams/delta_rds_request_additional_resources.svg
+   :alt: Delta RDS request additional resources
+
+Request Path Resolution
```

Comment:
This section seems less about the VHDS protocol and more about Envoy's internal implementation of VHDS.  I think that even if we keep the rest of the VHDS info in this doc, this part doesn't seem like it belongs here.

Reply:
As mentioned earlier, I believe there was a request from earlier on that implementation details were covered in the docs.

Reply:
I think @mattklein123 was driving that ask; we definitely need both types of docs. I'm in agreement that we can have the protocol docs be less proxy specific for the most part, the Envoy specifics could be treated in a proxy specific section.

Reply:
Where do you see this proxy specific documentation going? I think most of this is proxy-specific.

Reply:
I think Harvey meant "client-specific", not "proxy-specific".  Envoy is one such client, but my team is in the process of adding xDS support to gRPC, so that gRPC traffic can be configured via xDS without requiring use of a proxy.  If a gRPC client needs to get data from VHDS, then we need to speak the same protocol, but we won't use the client-side implementation that's being added to Envoy.

This is why I think it's important to define the protocol independently from the implementation.  gRPC would have its own xDS client implementation, and the protocol needs to be clearly defined in order for clients and management servers to interoperate.

Reply:
Let me know if this is resolved.

Reply:
Yes, client-specific.

Reply:
This looks good to me!